File: C#/Strategy/PaymentStrategys/VisaCardPaymentStrategy.cs
using Strategy.Interfaces;

namespace Strategy.PaymentStrategys;

public sealed class VisaCardPaymentStrategy : IPaymentStrategy
{
    /// <summary>
    /// Visa Card payment processing
    /// </summary>
    /// <param name="amount"><inheritdoc/></param>
    public void ProcessPayment(decimal amount)
        => Console.WriteLine($"Processing visa card payment for {amount}");
}

File: typescript/patterns/strategy/visa-card-payment-strategy.ts
import type { PaymentStrategy } from "./payment-strategy";

export class VisaCardPaymentStrategy implements PaymentStrategy {
  processPayment(amount: number): void {
    console.log("Processing payment of visa cards...");
    console.log(`Calculating fees of the amount ${amount} for visa cards...`);
  }
}
